olive/engine/engine.py

```diff
@@ -87,6 +90,38 @@ def __init__(
         else:
             self.target = LocalSystem()
 
+        if execution_providers is None:
+            execution_providers = self._config.execution_providers
+
+        # verfiy the AzureML system have specified the execution providers
+        # Please note we could not use isinstance(target, AzureMLSystem) since it would import AzureML packages.
+        if self.target.system_type == SystemType.AzureML and execution_providers is None:
+            raise ValueError("AzureMLSystem requires execution providers to be specified.")
+        elif execution_providers is None:
+            if self.target.system_type == SystemType.Local:
```

Comment:
Thanks for renaming the python environment system's `get_supported_execution_providers` method. I think now, we can add the python environment sustem to this `if` case too?